### PR TITLE
@damassi => Update acceptance tests to handle console errors

### PR DESCRIPTION
**Pull request by kanaabe**
Related convo here: https://artsy.slack.com/archives/C0350895E/p1501103674430615

WIP because there are two tests still spitting out error messages and I'm still trying to figure out the best place to fix them.

Also, I wanted to stub `app.get('/auctions/reminders')` after [this](https://github.com/artsy/force/blob/master/test/acceptance/helpers/index.js#L110) but because of express finding routes in order, I can't override it. I could recreate the app, but that feels like going down a rabbit hole just to stub one route hence all of the `stubAuctionReminder`s.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please assign someone to your PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/middleware/error_handler.coffee [hunk 1]
```diff
@@ -6,10 +6,10 @@ module.exports = (err, req, res, next) ->
...
-  console.log err if isNotProduction
+  isDevelopment = NODE_ENV is 'development'
+  console.log err if isDevelopment
   code = 504 if req.timedout
   code ||= err.status || 500
-  message = err.message || err.text || err.toString() if isNotProduction
-  detail = err.stack if isNotProduction
+  message = err.message || err.text || err.toString() if isDevelopment
+  detail = err.stack if isDevelopment
   res.status(code).render(file, { message, detail, code })
```
Comment (kanaabe) [inline lib/middleware/error_handler.coffee]: @damassi In trying to go back and stub routes in acceptance tests (see below for an example) to make error log outputs happy, I think the simplest solution is to log in dev mode only. If we're trying to test a 404, for example, there's no great way to suppress that error unless we mock console.log. This change should only affect acceptance tests anyway.

File: lib/middleware/error_handler.coffee [hunk 2]
```diff
@@ -6,10 +6,10 @@ module.exports = (err, req, res, next) ->
...
-  isNotProduction = NODE_ENV isnt 'production'
-  console.log err if isNotProduction
+  isDevelopment = NODE_ENV is 'development'
+  console.log err if isDevelopment
   code = 504 if req.timedout
   code ||= err.status || 500
-  message = err.message || err.text || err.toString() if isNotProduction
-  detail = err.stack if isNotProduction
+  message = err.message || err.text || err.toString() if isDevelopment
+  detail = err.stack if isDevelopment
```
Comment (damassi) [inline lib/middleware/error_handler.coffee]: 👍

**Comment by damassi** [comment]
👍  - Will merge on 🍏

**Comment by kanaabe** [comment]
Okie dokes, went for the re-pend. Going to merge this!

**Pull request merged**